File: ant-sdk/antd-py/examples/06_private_data.py
"""Example 06: Private (encrypted) data round-trip.

Private data is encrypted before storage. The returned data map
is required to retrieve and decrypt the data.
"""

from antd import AntdClient

client = AntdClient()

# Store private data
secret_message = b"This message is encrypted on the network"
result = client.data_put(secret_message)
data_map = result.data_map
print(f"Data map: {data_map}")
print(f"Chunks stored: {result.chunks_stored}, payment mode: {result.payment_mode_used}")

# Retrieve and decrypt
retrieved = client.data_get(data_map)
print(f"Decrypted: {retrieved.decode()}")

assert retrieved == secret_message, "Private data round-trip mismatch!"
print("Private data round-trip OK!")

File: ant-sdk/antd-js/examples/06-private-data.ts
/**
 * Example 06: Private (encrypted) data round-trip.
 *
 * Private data is encrypted before storage. The returned DataMap is
 * required to retrieve and decrypt the data and is NOT stored on-network.
 */

import { createClient } from "../src/index.js";

const client = createClient();

// Store private data
const secretMessage = Buffer.from("This message is encrypted on the network");
const result = await client.dataPut(secretMessage);
console.log(`Data map: ${result.dataMap}`);
console.log(`Chunks stored: ${result.chunksStored}, payment mode: ${result.paymentModeUsed}`);

// Retrieve and decrypt using the caller-held DataMap
const retrieved = await client.dataGet(result.dataMap);
console.log(`Decrypted: ${retrieved.toString()}`);

if (!retrieved.equals(secretMessage)) throw new Error("Private data round-trip mismatch!");
console.log("Private data round-trip OK!");
